From 1201c67c3f57b4030abd16a05ba50afcef833811 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Wed, 2 Sep 2015 19:45:35 +0300
Subject: [PATCH 0396/1240] mvebu: bubt: updated bubt command of SPI interface

- Updated bubt command of SPI interface to use SPI update function

Change-Id: I0306442e3e24639973084e2739de0e6087046191
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/23385
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
---
 common/cmd_sf.c         |  2 +-
 common/mvebu/cmd_bubt.c | 21 ++++-----------------
 include/spi_flash.h     |  1 +
 3 files changed, 6 insertions(+), 18 deletions(-)

diff --git a/common/cmd_sf.c b/common/cmd_sf.c
index 5c788e9..8c1aba7 100644
--- a/common/cmd_sf.c
+++ b/common/cmd_sf.c
@@ -202,7 +202,7 @@ static const char *spi_flash_update_block(struct spi_flash *flash, u32 offset,
  * @param buf		buffer to write from
  * @return 0 if ok, 1 on error
  */
-static int spi_flash_update(struct spi_flash *flash, u32 offset,
+int spi_flash_update(struct spi_flash *flash, u32 offset,
 		size_t len, const char *buf)
 {
 	const char *err_oper = NULL;
diff --git a/common/mvebu/cmd_bubt.c b/common/mvebu/cmd_bubt.c
index 2555c96..ca83a72 100644
--- a/common/mvebu/cmd_bubt.c
+++ b/common/mvebu/cmd_bubt.c
@@ -97,7 +97,7 @@ static ulong get_load_addr(void)
 #ifdef CONFIG_SPI_FLASH
 static int spi_burn_image(int image_size)
 {
-	int ret, erase_size;
+	int ret;
 	struct spi_flash *flash;
 
 	/* Probe the SPI bus to get the flash device */
@@ -112,25 +112,12 @@ static int spi_burn_image(int image_size)
 	spi_flash_protect(flash, 0);
 #endif
 
-	/* Erase the U-BOOT image space */
-	erase_size = ((image_size / flash->erase_size) + 1) * flash->erase_size;
-	printf("Erasing 0x%x - 0x%x:...", 0, erase_size);
-	ret = spi_flash_erase(flash, 0, erase_size);
-	if (ret) {
-		printf("Error!\n");
-		goto error;
-	}
-	printf("Done!\n");
-
-	/* Write the image tp flash */
-	printf("Writing image:...");
-	ret = spi_flash_write(flash, 0, image_size, (void *)get_load_addr());
+	ret = spi_flash_update(flash, 0, image_size, (void *)get_load_addr());
 	if (ret)
-		printf("\t\tError!\n");
+		printf("Error!\n");
 	else
-		printf("\t\tDone!\n");
+		printf("Done!\n");
 
-error:
 #ifdef CONFIG_SPI_FLASH_PROTECTION
 	spi_flash_protect(flash, 1);
 #endif
diff --git a/include/spi_flash.h b/include/spi_flash.h
index f4ae88d..7fdacf1 100644
--- a/include/spi_flash.h
+++ b/include/spi_flash.h
@@ -198,5 +198,6 @@ static inline int spi_flash_erase(struct spi_flash *flash, u32 offset,
 
 void spi_boot(void) __noreturn;
 void spi_spl_load_image(uint32_t offs, unsigned int size, void *vdst);
+int spi_flash_update(struct spi_flash *flash, u32 offset, size_t len, const char *buf);
 
 #endif /* _SPI_FLASH_H_ */
-- 
1.9.1

